templates/criterion.js

```diff
@@ -0,0 +1,870 @@
+    var outliers = report.reportAnalysis.anOutlierVar;
+    return elem('div', {className: 'outliers'}, [
+      elem('p', {}, [
+        'Outlying measurements have ',
+        outliers.ovDesc,
+        ' (', String((outliers.ovFraction * 100).toPrecision(3)), '%)',
+        ' effect on estimated standard deviation.'
+      ])
+    ]);
+  }
+
+  function mkTable(report) {
+    var analysis = report.reportAnalysis;
+    var timep4 = function(t) {
+      return formatTime(t, 3)
+    };
+    var idformatter = function(t) {
+      return t.toPrecision(3)
+    };
+    var rows = [
+      Object.assign({
+        label: 'OLS regression',
+        formatter: timep4
+      },
+        bounds(analysis.anRegress[0].regCoeffs.iters)),
+      Object.assign({
+        label: 'R² goodness-of-fit',
+        formatter: idformatter
+      },
+        bounds(analysis.anRegress[0].regRSquare)),
+      Object.assign({
+        label: 'Mean execution time',
+        formatter: timep4
+      },
+        bounds(analysis.anMean)),
+      Object.assign({
+        label: 'Standard deviation',
+        formatter: timep4
+      },
+        bounds(analysis.anStdDev)),
+    ];
+    return elem('table', {
+      className: 'analysis'
+    }, [
+      elem('thead', {}, [
+        elem('tr', {}, [
+          elem('th'),
+          elem('th', {
+            className: 'low',
+            title: confidence(analysis.anRegress[0].regCoeffs.iters.estError.confIntCL)
+          }, ['lower bound']),
+          elem('th', {}, ['estimate']),
+          elem('th', {
+            className: 'high',
+            title: confidence(analysis.anRegress[0].regCoeffs.iters.estError.confIntCL)
+          }, ['upper bound']),
+        ])
+      ]),
+      elem('tbody', {}, rows.map(function(row) {
+        return elem('tr', {}, [
+          elem('td', {}, [row.label]),
+          elem('td', {className: 'low'}, [row.formatter(row.low, 4)]),
+          elem('td', {}, [row.formatter(row.mean)]),
+          elem('td', {className: 'high'}, [row.formatter(row.high, 4)]),
+        ]);
+      }))
+    ]);
+  }
+  document.addEventListener('DOMContentLoaded', function() {
+    var reportData = JSON.parse(document.getElementById('report-data')
+      .getAttribute('data-report-json'))
+      .map(function(report) {
+        report.groups = report.reportName.split('/');
+        return report;
+      });
+    groupReports(reportData);
+    var overview = document.getElementById('overview-chart');
+    var overviewLineHeight = 16 * 1.25;
+    overview.style.height =
+      String(overviewLineHeight * reportData.length + 36) + 'px';
+    overview.appendChild(mkOverview(reportData.slice()));
+    var reports = document.getElementById('reports');
+    reportData.forEach(function(report, i) {
+      var id = 'report_' + String(i);
+      reports.appendChild(
+        elem('div', {id: id, className: 'report-details'}, [
+          elem('h1', {}, [elem('a', {href: '#' + id}, [report.groups.join(' / ')])]),
```

Comment:
Does this handle apostrophes in group names correctly? (See #202.)

Reply:
This creates a [text node](https://developer.mozilla.org/en-US/docs/Web/API/Document/createTextNode), which escapes HTML. However, it doesn't style the apostrophes in any specific way (by replacing with with i.e. a [right single quote](http://www.fileformat.info/info/unicode/char/2019/index.htm)). The issue in #202 should not be a problem here though.